raiden-dapp/src/service-worker/messages.ts

```diff
@@ -1,17 +1,90 @@
-/*
- * The naming of the message enumerations follows **who** is sending the
- * message. Also is it required to assign actual values to the messages to make
- * it work correctly within the service worker.
- */
-
-export enum ServiceWorkerMessageIdentifier {
+export enum ServiceWorkerMessageType {
   INSTALLED_VERSION = 'installed_version',
   INSTALLATION_ERROR = 'installation_error',
   RELOAD_WINDOW = 'reload_window',
   CACHE_IS_INVALID = 'cache_is_invalid',
-}
-
-export enum ServiceWorkerAssistantMessageIdentifier {
   UPDATE = 'update',
   VERIFY_CACHE = 'verify_cache',
+  UNKNOWN = 'unknown', // Used for messages which could not be parsed.
+}
+
+export type ServiceWorkerMessagePayload = Record<string, unknown>;
+
+export type ServiceWorkerMessageHandler = (payload: ServiceWorkerMessagePayload) => void;
+
+export abstract class ServiceWorkerMessage {
```

Comment:
Wow, this is a quite involved approach. I'd have preferred a simpler one, we don't expect these messages to change that often or again need to break backwards compatibility, but if it works, let's keep it like that

Reply:
Yes it is involved. I just want to solve this problem once and for all now. If we go with a cheap solution it becomes harder and harder each time. Because I missed the signs (hello we talk about messaging here, typical CS problem) it already includes the exception now. It would have become worse over time. Back then I also thought there will not much change to it. But it happened. So the best I can do is to keep the code open to changes I don't know about yet and might think they never happen. But as it is about compatibility here to old versions we can't touch anymore it is a serious business we can't just do when it really becomes super urgent/necessary.
Therefore I really like to keep this. As long as you think it does what it should (wrapped mapping with a week schema).